File: 3_LoginDemo/app/src/main/java/arjun/myappcompany/logindemo/MainActivity.java
package arjun.myappcompany.logindemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    public void onClickFunction(View view) {
        TextView usernameEditText = (TextView) findViewById(R.id.usernameEditText);
        TextView passwordEditText = (TextView) findViewById(R.id.passwordEditText);

        Log.i( "Username: ", usernameEditText.getText().toString());
        Log.i( "Password: ", passwordEditText.getText().toString());

        Toast.makeText( this, "Hi there!", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: 56_Snapchat/app/src/main/java/arjun/myappcompany/snapchat/MainActivity.kt
package arjun.myappcompany.snapchat

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.FirebaseDatabase

class MainActivity : AppCompatActivity() {

    var emailEditText: EditText? = null
    var passwordEditText: EditText? = null

    val mAuth = FirebaseAuth.getInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        emailEditText = findViewById(R.id.emailEditText)
        passwordEditText = findViewById(R.id.passwordEditText)

        if (mAuth.currentUser != null) {
            logIn()
        }
    }

    fun goClicked(view: View) {
        // Check if we can login the user
        mAuth.signInWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString())
            .addOnCompleteListener(this) { task ->
                if (task.isSuccessful) {
                   logIn()
                } else {
                    mAuth.createUserWithEmailAndPassword(emailEditText?.text.toString(), passwordEditText?.text.toString())
                        .addOnCompleteListener(this) { task ->
                            if (task.isSuccessful) {
                                task.result?.user?.let {
                                    FirebaseDatabase.getInstance().getReference().child("users").child(
                                        it.uid).child("email").setValue(emailEditText?.text.toString())
                                }
                                logIn()
                            } else {
                                Toast.makeText(this, "Login Failed. Try Again.",
                                    Toast.LENGTH_SHORT).show()
                            }
                        }
                }
            }
    }

    fun logIn() {
        // Move to next Activity
        val intent = Intent(this, SnapsActivity::class.java )
        startActivity(intent)
    }
}